rs/index/src/ivf/builder.rs

```diff
@@ -603,7 +590,27 @@ impl IvfBuilder {
             cur_idx += 1;
         }
 
-        Ok(())
+        Ok(cur_idx)
+    }
+
+    /// Assign new ids to the vectors
+    fn get_reassigned_ids(&mut self) -> Result<Vec<i32>> {
```

Comment:
function `get_reassigned_ids` shouldn't do the `assigning`. Let's require explicitly calling `reindex` call first

Reply:
the interface should be similar to hnsw

Reply:
I mean this is just to assign the IDs. The reindexing (rearranging vectors in vector storage + updating doc_id_mapping) will be done in `reindex` indeed.

This is the current structure of HNSW.